oss-licenses-plugin/src/main/groovy/com/google/android/gms/oss/licenses/plugin/DependencyUtil.groovy

```diff
@@ -0,0 +1,224 @@
+                continue
+            }
+
+            def resolvedDependencies = configuration.getResolvedConfiguration().getLenientConfiguration().allModuleDependencies
+            return findLibraryFileInResolvedDependencies(resolvedDependencies, artifactInfo, Set.of())
+        }
+        logger.warn("No resolved configurations contained $artifactInfo")
+        return null
+    }
+
+    private static File findLibraryFileInResolvedDependencies(
+            Set<ResolvedDependency> resolvedDependencies,
+            ArtifactInfo artifactInfo,
+            Set<ResolvedDependency> visitedDependencies) {
+
+        for (ResolvedDependency resolvedDependency in resolvedDependencies) {
+            try {
+                if (resolvedDependency.getModuleVersion() == LOCAL_LIBRARY_VERSION) {
+                    /**
+                     * Attempting to getAllModuleArtifacts on a local library project will result
+                     * in AmbiguousVariantSelectionException as there are not enough criteria
+                     * to match a specific variant of the library project. Instead we skip the
+                     * the library project itself and enumerate its dependencies.
+                     */
+                    if (resolvedDependency in visitedDependencies) {
+                        logger.info("Dependency cycle at ${resolvedDependency.name}")
+                        continue
+                    }
+                    File childResult = findLibraryFileInResolvedDependencies(
+                            resolvedDependency.getChildren(),
+                            artifactInfo,
+                            visitedDependencies + resolvedDependency
+                    )
+                    if (childResult != null) {
+                        return childResult
+                    }
+                } else {
+                    for (resolvedArtifact in resolvedDependency.getAllModuleArtifacts()) {
+                        if (isMatchingArtifact(resolvedArtifact, artifactInfo)) {
+                            return resolvedArtifact.file
+                        }
+                    }
+                }
+            } catch (AmbiguousVariantSelectionException exception) {
+                logger.info("Failed to process ${resolvedDependency.name}", exception)
+            }
+        }
+        return null
+    }
+
+    /**
+     * Returns true for configurations that cannot be resolved in the newer
+     * version of gradle API, are tests, or are not packaged dependencies.
+     */
+    private static boolean shouldSkipConfiguration(Configuration configuration) {
+        (!canBeResolved(configuration)
+                || isTest(configuration)
+                || !isPackagedDependency(configuration))
+    }
+
+    /**
+     * Checks if the configuration can be resolved. isCanBeResolved is added to
+     * gradle Api since 3.3. For the previous version, all the configurations
```

Comment:
do you still support such old Gradle version ? 
I don't even remember when AGP required 3.3 :)

Reply:
Now that we require AGP 7.1 we'll never see Gradle that old, but I was trying to leave this older code largely untouched in the refactor.